R/carehomes_rt.R

```diff
@@ -63,6 +63,39 @@ carehomes_Rt <- function(step, S, p, prob_strain = NULL,
                          eigen_method = "power_iteration", R = NULL,
                          weight_Rt = FALSE) {
 
+  ## deal with the prob_strain NA case at the beginning before any variables
+  ##  are modified
+  if (!is.null(prob_strain) && any(is.na(prob_strain))) {
+    which_nna <- !vapply(seq(ncol(prob_strain)),
+                        function(i) any(is.na(prob_strain[, i])), logical(1))
+
+    ## calculate Rt by first ignoring NA
+    ret <- carehomes_Rt(step[which_nna], S[, which_nna], p,
+                        prob_strain[, which_nna], type, interpolate_every,
+                        interpolate_critical_dates, interpolate_min,
+                        eigen_method, R[, which_nna], weight_Rt)
+
+    ## replace reduced step,date,beta with full values (no NA here)
+    ret$step <- step
+    ret$date <- step * p$dt
+    ret$beta <- beta <- sircovid_parameters_beta_expand(step, p$beta_step)
+
+    ## restore full length Rt with NAs when prob_strain is NA
+    for (i in 4:length(ret)) {
```

Comment:
`for (i in type) {` ?

Reply:
Doesn't work because `type` may still be `NULL` at this point in the code. I could move it below L109 to resolve this but feel this is easier to read

Reply:
Can we move the `type` resolution above this then? the `4:` make me a bit queasy but it's not a huge deal

Reply:
@richfitz  and @RaphaelS1  please check you are happy with my proposed change for this?